### DISCUSSION TITLE
pin.forwardKinematics and pin.updateFramePlacements

### DISCUSSION THREAD

**Discussion by LHT-Liu** [stack-of-tasks/pinocchio#2215] [Q&A]
Hello, I want to get the pose of a joint relative to the world coordinate system, and then use `pin.forwardKinematics(model, data, q0)` and `pose = data.oMi[Joint_id]`, but I noticed that there are two other output methods,
![image](https://github.com/stack-of-tasks/pinocchio/assets/64388319/304a24c0-2600-456c-8d00-c74d1aa73c7d)
How to understand the speed and acceleration input parameters in these two methods？

I want to get the pose of frame (such as fixed joint), add `pin.updateFramePlacements(model, data)` after `pin.forwardKinematics(model, data, q0)`, and then use `pose = data.oMf[frame_id]` to get it, right?

@jviereck @stephane-caron

**Comment by stephane-caron** [comment]
> I want to get the pose of frame (such as fixed joint), add pin.updateFramePlacements(model, data) after pin.forwardKinematics(model, data, q0), and then use pose = data.oMf[frame_id] to get it, right?

Yes, that is the way to go to get the pose of a frame.

The other two methods are useful respectively if you want to know the velocity of a frame (corresponding to a given `q` and `v`), or the acceleration of a frame (corresponding to a given `q`, `v` and `a`).
